cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowDispatch.qll

```diff
@@ -135,6 +135,12 @@ private module VirtualDispatch {
       exists(FunctionInstruction fi |
         this.flowsFrom(DataFlow::instructionNode(fi), _) and
         result = fi.getFunctionSymbol()
+      ) and
+      (
+        this.getNumberOfArguments() <= result.getEffectiveNumberOfParameters() and
```

Comment:
Why not just a plain `=`? Does this avoid making the compiler emit a join, and thus avoid an unfortunate join order?

Reply:
Yep, it fixes a join ordering problem where it would join together all calls to all functions with the right number of arguments